src/webhooks/utils/handleOldInvoiceCompletedFlow.ts

```diff
@@ -0,0 +1,81 @@
+import Stripe from 'stripe';
+import { AppConfig } from '../../config';
+import { createOrUpdateUser, updateUserTier } from '../../services/storage.service';
+import { FastifyBaseLogger } from 'fastify';
+import { User } from '../../core/users/User';
+import { UsersService } from '../../services/users.service';
+import { NoSubscriptionSeatsProvidedError } from '../../services/tiers.service';
+
+interface HandleOldInvoiceCompletedFlowProps {
+  maxSpaceBytes: string;
+  userUuid: User['uuid'];
+  isBusinessPlan: boolean;
+  subscriptionSeats: number | null;
+  customer: Stripe.Customer;
+  config: AppConfig;
+  product: Stripe.Product;
+  usersService: UsersService;
+  log: FastifyBaseLogger;
+}
+
+export const handleOldInvoiceCompletedFlow = async ({
+  maxSpaceBytes,
+  isBusinessPlan,
+  usersService,
+  userUuid,
+  subscriptionSeats,
+  customer,
+  config,
+  product,
+  log,
+}: HandleOldInvoiceCompletedFlowProps) => {
+  if (isBusinessPlan) {
+    if (!subscriptionSeats)
+      throw new NoSubscriptionSeatsProvidedError('The amount of seats is not allowed for this type of subscription');
+
+    const address = customer.address?.line1 ?? undefined;
+    const phoneNumber = customer.phone ?? undefined;
+
+    try {
+      await usersService.updateWorkspaceStorage(userUuid, Number(maxSpaceBytes), subscriptionSeats);
+      log.info(
+        `USER WITH CUSTOMER ID: ${customer.id} - UUID: ${userUuid} - EMAIL: ${
+          customer.email
+        } HAS BEEN UPDATED HIS WORKSPACE`,
+      );
+    } catch (err) {
+      const error = err as Error;
+      const statusCode = (err as any)?.response.status;
+
+      if (!statusCode || statusCode !== 404) {
+        log.error(`[ERROR UPDATING WORKSPACE]: ${error.stack ?? error.message}`);
+        throw err;
+      }
+
+      log.info(
+        `USER WITH CUSTOMER ID: ${customer.id} - UUID: ${userUuid} - EMAIL: ${
+          customer.email
+        } DOES NOT HAVE ANY WORKSPACE TO UPDATE, CREATING A NEW ONE`,
+      );
+      await usersService.initializeWorkspace(userUuid, {
+        newStorageBytes: Number(maxSpaceBytes),
```

Comment:
Shouldn't the newStorageBytes be `maxSpaceBytes` * `subscriptionSeats` or am I missing something ?

Reply:
I just copied/pasted the logic we had in the handler. I guess this operation will be done in the drive-server side.